internal/ratelimit/monitor.go

```diff
@@ -180,6 +181,17 @@ func (c *Monitor) Update(h http.Header) {
 	c.limit = limit
 	c.remaining = remaining
 	c.reset = time.Unix(resetAtSeconds, 0)
+
+	if c.collector != nil {
+		c.collector(c.known, float64(c.remaining))
+	}
```

Comment:
Instead of threading through this `known` boolean just to early return in the collector itself, we can just test for that value here?
```suggestion
	if c.collector != nil && c.known {
		c.collector(float64(c.remaining))
	}
```

Reply:
Good idea